### ISSUE TITLE
0.13 not honor filesystem_mirror in .terraformrc settings after switching workspace

### ISSUE THREAD

**Issue by DesmondH0** [hashicorp/terraform#25794]
### TLDR
There are technically 3 issues after upgrading to 0.13:
1. Terraform required to re-init after switching workspace. 
2. Terraform redownload the modules and put them into `.terraform/plugins/-/` after switching workspace.  
    For `default` workspace, modules are downloaded to `.terraform/plugins/`  
3. After switching workspace beside of default, Terraform do not honor the settings in `.terraformrc`, so `filesystem_mirror` not working anymore

### Terraform Version
Terraform v0.13.0

### Terraform Configuration Files
`~/.terraformrc`:  
```
provider_installation {
  filesystem_mirror {
    path    = "/terraform/providers"
    include = ["example.com/*/*"]
  }
  direct {
    exclude = ["example.com/*/*"]
  }
}
```

`version.tf`:  
```
terraform {
  required_providers {
    google = {
      source = "hashicorp/google"
      version = "~> 3.15.0"
    }
    ....
    kubectl = {
      source = "example.com/terraform/kubectl"
      version = "~> 1.6.2"
    }
    ....
  }
  required_version = ">= 0.13"
}
```

### Debug Output
In Default workspace:  
```
desmondho$ terraform init

Initializing the backend...

Initializing provider plugins...
- Using previously-installed hashicorp/google v3.15.0
- Using previously-installed hashicorp/kubernetes v1.11.4
- Using previously-installed hashicorp/helm v1.1.1
- Using previously-installed hashicorp/template v2.1.2
- Using previously-installed terraform-providers/okta v3.2.0
- Using previously-installed example.com/terraform/kubectl v1.6.2
- Using previously-installed hashicorp/tls v2.1.1
- Using previously-installed hashicorp/google-beta v3.15.0
- Using previously-installed hashicorp/vault v2.12.2

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
```

In other workspace:
```
Error: Could not load plugin

Plugin reinitialization required. Please run "terraform init".
```

When trying to do the `terraform init`, it went to official registry instead:  
```
Initializing the backend...

Initializing provider plugins...
(same output as default...)
- Finding latest version of -/kubectl...

Error: Failed to query available provider packages

Could not retrieve the list of available versions for provider -/kubectl:
provider registry registry.terraform.io does not have a provider named
registry.terraform.io/-/kubectl
```

### Steps to Reproduce
0. from any workable module
1. using the `versions.tf` and `~/.terraformrc` above
2. download the `kubectl` 3rd party module from [here](https://github.com/gavinbunney/terraform-provider-kubectl/releases) and put it into `/terraform/providers/example.com/terraform/kubectl`
3. terraform workspace select default
4. terraform init (This command should work)
5. terraform workspace new testing
6. terraform init (This command failed)

**Comment by tinder-tder** [comment]
The issue is your state file refrences the old unpathed provider
looking at the 013 upfrade doc they have a section on "terraform state replace-provider".  I ran into the same issue
running:
terraform state replace-provider 'registry.terraform.io/-/kubectl' 'registry.terraform.io/gavinbunney/kubectl'

fixed it for me, you can change the kubectl path to whatever

**Comment by DesmondH0** [comment]
@tinder-tder , thanks!
i was missed that part after upgrade. seems working with `terraform state replace-provider` command

closing this now
